logger: allow logging HTTP traffic w/o logging everything

MiguelPires opened the pull request:
Previously, for anything to be logged SNAPD_DEBUG had to be set to a true value (1, true, etc) even if SNAPD_DEBUG_HTTP was already set. In practice, this meant that we couldn't get HTTP logs w/o getting all other logging. This changes that so we can log only HTTP traffic by setting SNAPD_DEBUG_HTTP.

mvo5 reviewed (approved):
Thank you, this looks good

pedronis commented:
> This implies that "normal" snapd debug info will also be shown even if only SNAPD_DEBUG_HTTP is set but

this sounds more confusing than the status quo to me

MiguelPires commented:
@pedronis thanks for the review, changed this to enable just logging HTTP traffic w/o implying or requiring SNAPD_DEBUG to be set. Changed the title and description as well

pedronis commented on logger/logger.go:
```diff
@@ -37,6 +37,9 @@ type Logger interface {
 	Debug(msg string)
+	// DebugAlways is for messages that we always want to print so the gating must
+	// be done by the caller
+	DebugAlways(msg string)
```
Wondering about this name, mostly also because "DebugAlwaysf" is strange,  I don't have strong opinions but other options could be:

NoGuardDebug
DebugNotice
DebugLog

pedronis reviewed (approved):
thank you, some wonderings about the new Logger method name, especially when suffix with "f".

pedronis commented:
@MiguelPires question: is there some docs for this that need to change?

mvo5 commented on logger/logger.go:
```diff
@@ -37,6 +37,9 @@ type Logger interface {
 	// Debug is for messages that the user should be able to find if they're debugging something
 	Debug(msg string)
+	// NoGuardDebug is for messages that we always want to print (e.g., debug checks
+	// were done by the caller
```
(nitpick^2) there is a `)` missing

MiguelPires commented:
@pedronis the HACKING.md didn't specify that you needed to turn on both env vars so it's not outdated. I can add something to it so we're explicit about how it works

mvo5 reviewed (approved):
Thank you

MiguelPires commented:
> @pedronis the HACKING.md didn't specify that you needed to turn on both env vars so it's not outdated. I can add something to it so we're explicit about how it works

Actually, I'll wait until the PR with the new env var is merged and then I'll update the HACKING to reflect both
